HopsanGUI/Widgets/ScriptEditor.cpp

```diff
@@ -0,0 +1,407 @@
+/*-----------------------------------------------------------------------------
+
+ Copyright 2017 Hopsan Group
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ The full license is available in the file GPLv3.
+ For details about the 'Hopsan Group' or information about Authors and
+ Contributors see the HOPSANGROUP and AUTHORS files that are located in
+ the Hopsan source code root directory.
+
+-----------------------------------------------------------------------------*/
+
+//!
+//! @file   ScriptEditor.h
+//! @author Robert Braun <robert.braun@liu.se>
+//! @date   2018-10-01
+//! @version $Id$
+//!
+//! @brief Contains the script editor class
+//!
+
+
+#include "ScriptEditor.h"
+#include "Utilities/HighlightingUtilities.h"
+#include "global.h"
+#include "Configuration.h"
+#include "MessageHandler.h"
+#include "Widgets/ProjectTabWidget.h"
+#include "HcomHandler.h"
+#include "Widgets/HcomWidget.h"
+
+#include <QGridLayout>
+#include <QFileDialog>
+#include <QStringListModel>
+#include <QAbstractItemView>
+#include <QScrollBar>
+#include <QTextDocumentFragment>
+#include <QKeyEvent>
+#include <QPrinter>
+#include <QPainter>
+#include <QPrintDialog>
+#include <math.h>
+
+ScriptEditor::ScriptEditor(QFileInfo scriptFileInfo, QWidget *parent) : QWidget(parent)
+{
+    mScriptFileInfo = scriptFileInfo;
+
+    mpEditor = new HcomEditor(this);
+    QFont font("Monospace");
+    font.setStyleHint(QFont::TypeWriter);
+    mpEditor->setFont(font);
+
+    HcomHighlighter *pHighLighter = new HcomHighlighter(mpEditor->document());
+
+    if(mScriptFileInfo.exists())
+    {
+        QFile scriptFile(mScriptFileInfo.absoluteFilePath());
+        if(!scriptFile.open(QFile::ReadOnly | QFile::Text))
+        {
+            gpMessageHandler->addErrorMessage("Unable to read from HCOM script file: "+mScriptFileInfo.absoluteFilePath());
+            return;
+        }
+        mSavedText = scriptFile.readAll();
+        mpEditor->setText(mSavedText);
+        scriptFile.close();
+    }
+
+    QGridLayout *pLayout = new QGridLayout(this);
+    pLayout->addWidget(mpEditor,         0,0,1,2);
+    this->setLayout(pLayout);
+
```

Comment:
I dont think that you need to set the layout. Since this was the parent. In debug mode annoying warnings are shown that "widget already have a layout"